extensions/agent/src/mainController.ts

```diff
@@ -91,6 +124,117 @@ export class MainController {
+				the second argument is undefined
+			2. Notebook toolbar:
+				both the args are undefined
+			3. Agent New Notebook Action
+				the first arg is database OwnerUri
+				the second arg is undefined
+			4. Agent Edit Notebook Action
+				the first arg is database OwnerUri
+				the second arg is notebookInfo from database
+			*/
+			if (!ownerUri || ownerUri instanceof vscode.Uri) {
+				let path: string;
+				if (!ownerUri) {
+					if (azdata.nb.activeNotebookEditor.document.isDirty) {
+						vscode.window.showErrorMessage(localize('agent.unsavedFileSchedulingError', 'Save file before scheduling'), { modal: true });
+						return;
+					}
+					path = azdata.nb.activeNotebookEditor.document.fileName;
+				} else {
+					path = ownerUri.fsPath;
+				}
+
+				let connection = await this.getConnectionFromUser();
+				ownerUri = await azdata.connection.getUriForConnection(connection.connectionId);
+				this.notebookDialog = new NotebookDialog(ownerUri, <NotebookDialogOptions>{ filePath: path, connection: connection });
+				if (!this.notebookDialog.isOpen) {
+					this.notebookDialog.dialogName ? await this.notebookDialog.openDialog(this.notebookDialog.dialogName) : await this.notebookDialog.openDialog();
+				}
+			}
+			else {
+				if (!this.notebookDialog || (this.notebookDialog && !this.notebookDialog.isOpen)) {
+					this.notebookDialog = new NotebookDialog(ownerUri, <NotebookDialogOptions>{ notebookInfo: notebookInfo });
+				}
+				if (!this.notebookDialog.isOpen) {
+					this.notebookDialog.dialogName ? await this.notebookDialog.openDialog(this.notebookDialog.dialogName) : await this.notebookDialog.openDialog();
+				}
+			}
+		});
+	}
+
+	public async getConnectionFromUser(): Promise<azdata.connection.Connection> {
+		let connection: azdata.connection.Connection = null;
+
+		let connections = await azdata.connection.getActiveConnections();
+		if (!connections || connections.length === 0) {
+			connection = await azdata.connection.openConnectionDialog();
+		}
+		else {
+			let sqlConnectionsPresent: boolean;
+			for (let i = 0; i < connections.length; i++) {
+				if (connections[i].providerName === 'MSSQL') {
+					sqlConnectionsPresent = true;
+					break;
+				}
+			}
+			let connectionNames: azdata.connection.Connection[] = [];
+			let connectionDisplayString: string[] = [];
+			for (let i = 0; i < connections.length; i++) {
+				let currentConnectionString = connections[i].options.server + ' (' + connections[i].options.user + ')';
+				connectionNames.push(connections[i]);
+				connectionDisplayString.push(currentConnectionString);
+			}
+			connectionDisplayString.push(localize('agent.AddNewConnection', 'Add new connection'));
+			let connectionName = await vscode.window.showQuickPick(connectionDisplayString, { placeHolder: localize('agent.selectConnection', 'Select a connection') });
+			if (connectionDisplayString.indexOf(connectionName) !== -1) {
+				if (connectionName === 'Add new connection') {
```

Comment:
This is incorrect since you localize the string above. If someone is using a language other than English, this will fail. Please compare it to the localized string.